src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Main object to be created to initialise the chatbot program.
+ * 
+ * @author Siau Wee
+ */
+public class Duke {
+
+    private Ui ui;
+
+    private Parser parser;
+
+    private Storage storage;
+
+    private TaskList taskList;
```

Comment:
```suggestion
    private final Ui ui;

    private final Parser parser;

    private final Storage storage;

    private final TaskList taskList;
```
Suggesting the addition of the `final` modifier.